learn/getting-started/writing-a-graphql-api-with-ballerina.md

```diff
@@ -0,0 +1,397 @@
+isolation, etc.
+
+This tutorial includes the following steps:
+
+1. Design the GraphQL endpoint
+2. Create the Covid-19 dataset
+3. Write the GraphQL service to:
+    - Get all the Covid-19 data
+    - Filter Covid-19 data using the `isoCode`
+    - Add Covid-19 data
+
+## Prerequisites
+
+- Ballerina SwanLake installation (To download Ballerina, visit [Download Ballerina](https://ballerina.io/downloads/))
+- A tool to edit the code. (VSCode with Ballerina plugin is recommended. To download,
+  visit [Ballerina VSCode Plugin](https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina))
+- A command terminal.
+
+## Design the GraphQL endpoint
+
+Usually, a GraphQL endpoint is defined using a GraphQL schema. Some languages require the GraphQL schema to create a
+GraphQL service (schema-first approach), while some languages does not need the schema to create the service.
+(code-first approach). Ballerina GraphQL package uses the latter. Therefore, we do not need the schema file to create
+our service. But for the sake of the design, let's look at the GraphQL schema.
+
+```graphql
+type Query {
+    allData: [CovidData!]!
+    filterData(isoCode: String!): CovidData
+}
+
+type Mutation {
+    addData(entry: CovidEntry!): CovidData!
+}
+
+scalar Decimal
+
+type CovidData {
+    isoCode: String!
+    country: String!
+    recovered: Decimal
+    cases: Decimal
+    active: Decimal
+    deaths: Decimal
+}
+
+input CovidEntry {
+    isoCode: String!
+    country: String!
+    recovered: Decimal
+    cases: Decimal
+    active: Decimal
+    deaths: Decimal
+}
+```
+
+> Note: The `Decimal` type used here is coming from the `decimal` type in Ballerina language. We are using this type
+> to store the case counts. More about this later.
+
+Now that we have the design, we can create the project.
+
+## Create a Ballerina Project
+
+1. Open a terminal and move to the directory where you want to create the Ballerina project.
+
+   On Linux or Mac:
+    ```shell
+    cd your/parent/directory/path
+    ```
+
+   On Windows:
+    ```windows
+    cd your\parent\directory\path
+    ```
+2. Using the `bal` command, create a new Ballerina project.
+    ```shell
+    bal new covid19
+   ```
+This will create a new Ballerina project inside a directory named `covid19`.
+
+## Create a Datasource for the Project
+
+Before writing the GraphQL service, we are going to create a data source for out project. This will mimic a database
+where we store the data for our service. Use the following command to create a new module inside the Ballerina project.
+
+```shell
+bal add datasource
+```
+
+This will create a new directory named `modules/datasource`, which will be the Ballerina module acting as the data
+source of the GraphQL service.
+
+Now, let's add the data needed for the GraphQL service inside this module.
+
+First, we have to define our data types. We can use Ballerina records for this. Let's first define a Ballerina record to
```

Comment:
```suggestion
First, define the data types using Ballerina records. Let's first define a Ballerina record to
```
Rather than saying `first`, can we add these as numbered bullet points of this represents a sequence of steps/instructions?